### PR TITLE
feat(card_image): Improve `card_image()` API and usage

### LINKED ISSUE #1071: `card_image()` should include explicit `alt` argument
This would encourage users to thing about the accessibility of their cards.

### PR DESCRIPTION
**Pull request by gadenbuie**
This PR improves the `card_image()` API and fixes a few minor rough edges, including fixing #1071.

* Currently `card_image()` is by default wrapped in a `card_body()` container. This happens even if setting `container = NULL` because the `card_image()` result doesn't have a `card_item` class. This makes it impossible to use `card_image()` for [the card image caps described in the Boostrap docs](https://getbootstrap.com/docs/5.3/components/card/#images).

    With this PR, we now default to `container = NULL` and, when this is the case, call `as.card_item()` on the `image` so that the image is treated as a card item.

    I'm not sure how much value `container` adds -- it's easy enough to do `card_body(card_image())` if that's really desired -- and I don't think there's any harm in unconditionally adding the `card_item` class to the `image`. I'd be in favor of deprecating the `container` argument (unless I've missed an important use case).

* `border_radius` gains `"auto"`, the new default value. When `border_radius = "auto"`, we add a `card_image_auto` class to the `card_image()` output that is detected by `card()`. We then automatically add the `card-img-top`, `card-img-bottom`, or `card-img` class if the card image is the first, last or only child. In all other cases, we don't modify the image tag.

* I changed the default value of `card_image(fill = )` to `FALSE`. `fill = TRUE` might have worked better when `card_image()` was always wrapped in a `card_body()` but in most cases having `fill = TRUE` will cause the image to stretch, distorting the image. I don't think this is what most people want from `card_image()`.

* I added a bit of logic such that a URI or non-local file path can be given to `file`. Instead of having to write

    ```r
    card_image(file = NULL, src = "https://example.com/cat.png")
    ```

    users can just give the URL to `file`

    ```r
    card_image("https://example.com/cat.png")
    ```

    This is backwards compatible in the sense that when a value is passed to `src`, that value is used directly. The advantage is that we won't even consulte `file` when `src` is provided, allowing `card_image(src = "cat.png")` to skip the base64 encoding.

* I added `alt = ""` as a default argument that appears before the dots. This has the side effect of marking most `card_image()` uses as decorative. That's probably better than the default of not having alt text at all, and it intends to bring increased attention to alt text without breaking existing apps by making `alt` required. Fixes #1071

<details><summary>Example app</summary>

```r
library(shiny)
# library(bslib)
pkgload::load_all()

card_img <- card_image(
  file = "https://picsum.photos/seed/posit/600/300"
)

ui <- page_fillable(
  layout_column_wrap(
    width = 300,
    # advanced usage with a card image overlay
    # see: https://getbootstrap.com/docs/5.3/components/card/#image-overlays
    card(
      class = "text-light",
      card_image(
        file = "https://picsum.photos/seed/posit/600/900",
        # fill = TRUE,
        border_radius = "all"
      ),
      card_body(
        class = "card-img-overlay",
        lorem::ipsum(2)
      )
    ),
    # Typical usage: card top cap
    card(
      card_img,
      lorem::ipsum(1),
    ),
    # Typical usage: card bottom cap
    card(
      lorem::ipsum(1),
      card_img,
    ),
    # Typical usage: images for top/bottom cap, plus one in middle
    card(
      fill = FALSE,
      card_img,
      lorem::ipsum(1, 2),
      card_img,
      lorem::ipsum(1, 2),
      card_img
    )
  )
)

server <- function(input, output, session) {

}

shinyApp(ui, server)
```

</details>

image

### CODE REVIEW

File: R/card.R [hunk 1]
```diff
@@ -90,6 +90,7 @@ card <- function(
 
   attribs <- args[nzchar(argnames)]
   children <- as_card_items(args[!nzchar(argnames)], wrapper = wrapper)
+  children <- card_image_add_classes(children)
```
Comment (cpsievert) [inline R/card.R:93]: Let's investigate what this would mean for `navset_card_*()`

File: R/card.R [hunk 2]
```diff
@@ -248,32 +249,78 @@ card_footer <- function(..., class = NULL) {
...
-  mime_type = NULL, class = NULL, height = NULL, fill = TRUE, width = NULL, container = card_body) {
-
-  src <- NULL
-  if (length(file) > 0) {
-    src <- base64enc::dataURI(
-      file = file, mime = mime_type %||% mime::guess_type(file)
-    )
+  file,
+  alt = "",
+  ...,
```
Comment (cpsievert) [inline R/card.R:294]:
```suggestion
  ...,
  alt = "",
```

Comment (cpsievert) [reply R/card.R:294]: I would also be open to throwing an (informative) warning if unnamed args are found.

**Review by cpsievert — APPROVED** [review]

File: R/card.R (hunk 1 above)
Comment (gadenbuie) [reply R/card.R:93]: Turns out this works perfectly with `navset_card_*()`.


https://github.com/user-attachments/assets/fe485a18-fa0e-4ffa-bd9b-a9c7fdbb2395

**Pull request merged**